libs/libc/net/lib_getifaddrs.c

```diff
@@ -0,0 +1,275 @@
+  char                    name[IF_NAMESIZE];
+  struct sockaddr_storage addr;
+  struct sockaddr_storage netmask;
+  struct sockaddr_storage dstaddr;
+  struct sockaddr         hwaddr;
+};
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: getifaddrs
+ *
+ * Description:
+ *   The getifaddrs() function returns a link list of ifaddrs structures,
+ *   each containing information about one of the network interfaces on the
+ *   local system. The ifaddrs structure contains at least the following
+ *   entries:
+ *        struct ifaddrs  *ifa_next;
+ *        char            *ifa_name;
+ *        unsigned int     ifa_flags;
+ *        struct sockaddr *ifa_addr;
+ *        struct sockaddr *ifa_netmask;
+ *        struct sockaddr *ifa_dstaddr;
+ *        void            *ifa_data;
+ *  The ifa_next field contains a pointer to the next structure on
+ *  the list, or NULL if this is the last item of the list.
+ *
+ *  The ifa_name points to the null-terminated interface name.
+ *
+ *  The ifa_flags field contains the interface flags, as returned by
+ *  the SIOCGIFFLAGS ioctl(2) operation (see netdevice(7) for a list
+ *  of these flags).
+ *
+ *  The ifa_addr field points to a structure containing the interface
+ *  address.  (The sa_family subfield should be consulted to
+ *  determine the format of the address structure.)  This field may
+ *  contain a null pointer.
+ *
+ *  The ifa_netmask field points to a structure containing the
+ *  netmask associated with ifa_addr, if applicable for the address
+ *  family.  This field may contain a null pointer.
+ *
+ *  Depending on whether the bit IFF_BROADCAST or IFF_POINTOPOINT is
+ *  set in ifa_flags (only one can be set at a time), either
+ *  ifa_broadaddr will contain the broadcast address associated with
+ *  ifa_addr (if applicable for the address family) or ifa_dstaddr
+ *  will contain the destination address of the point-to-point
+ *  interface.
+ *
+ *  The ifa_data field points to a buffer containing address-family-
+ *  specific data; this field may be NULL if there is no such data
+ *  for this interface.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   On success, getifaddrs() returns pointer to the link list; on error,
+ *   NULL is returned, and errno is set to indicate the error.
+ *
+ ****************************************************************************/
+
+int getifaddrs(FAR struct ifaddrs **addrs)
+{
+  FAR struct myifaddrs *myaddrs = NULL;
+  int sockfd;
+  int i;
+
+  if (addrs == NULL)
+    {
+      set_errno(EINVAL);
+      return ERROR;
+    }
+
+  sockfd = socket(NET_SOCK_FAMILY, NET_SOCK_TYPE, NET_SOCK_PROTOCOL);
+  if (sockfd < 0)
+    {
+      return sockfd;
+    }
+
+  for (i = 1; i <= MAX_IFINDEX; i++)
```

Comment:
```suggestion
  for (i = 1, *addrs = NULL; i <= MAX_IFINDEX; i++)
```
Just to prevent if some garbage is pointed by `addrs`

Reply:
Done.